main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidator.java

```diff
@@ -32,56 +34,101 @@
  *       or departure_time
+ *   <li>{@link MissingTimepointValueNotice} - value for {@code stop_times.timepoint} is missing
+ *   <li>{@link MissingTimepointColumnNotice} - field {@code stop_times.timepoint} is missing
  * </ul>
  */
 @GtfsValidator
-public class TimepointTimeValidator extends SingleEntityValidator<GtfsStopTime> {
+public class TimepointTimeValidator extends FileValidator {
+  private final GtfsStopTimeTableContainer stopTimes;
+
+  @Inject
+  TimepointTimeValidator(GtfsStopTimeTableContainer stopTimes) {
+    this.stopTimes = stopTimes;
+  }
 
   @Override
-  public void validate(GtfsStopTime stopTime, NoticeContainer noticeContainer) {
-    if (!isTimepoint(stopTime)) {
+  public void validate(NoticeContainer noticeContainer) {
+    if (!stopTimes.hasColumn(GtfsStopTimeTableLoader.TIMEPOINT_FIELD_NAME)) {
+      // legacy datasets do not use timepoint column in stop_times.txt as a result:
+      // - this should be flagged;
+      // - but also no notice regarding the absence of arrival_time or departure_time should be
+      // generated
+      noticeContainer.addValidationNotice(new MissingTimepointColumnNotice());
       return;
     }
-    if (!stopTime.hasArrivalTime()) {
-      noticeContainer.addValidationNotice(
-          new StopTimeTimepointWithoutTimesNotice(
-              stopTime.csvRowNumber(),
-              stopTime.tripId(),
-              stopTime.stopSequence(),
-              String.format(GtfsStopTimeTableLoader.ARRIVAL_TIME_FIELD_NAME)));
-    }
-    if (!stopTime.hasDepartureTime()) {
-      noticeContainer.addValidationNotice(
-          new StopTimeTimepointWithoutTimesNotice(
-              stopTime.csvRowNumber(),
-              stopTime.tripId(),
-              stopTime.stopSequence(),
-              GtfsStopTimeTableLoader.DEPARTURE_TIME_FIELD_NAME));
+    for (GtfsStopTime stopTime : stopTimes.getEntities()) {
+      if (!stopTime.hasTimepoint()) {
+        noticeContainer.addValidationNotice(new MissingTimepointValueNotice(stopTime));
+      }
+      if (isTimepoint(stopTime)) {
+        if (!stopTime.hasArrivalTime()) {
+          noticeContainer.addValidationNotice(
+              new StopTimeTimepointWithoutTimesNotice(
+                  stopTime, GtfsStopTimeTableLoader.ARRIVAL_TIME_FIELD_NAME));
+        }
+        if (!stopTime.hasDepartureTime()) {
+          noticeContainer.addValidationNotice(
+              new StopTimeTimepointWithoutTimesNotice(
+                  stopTime, GtfsStopTimeTableLoader.DEPARTURE_TIME_FIELD_NAME));
+        }
+      }
     }
   }
 
   private boolean isTimepoint(GtfsStopTime stopTime) {
-    return stopTime.timepoint().equals(GtfsStopTimeTimepoint.EXACT);
+    return stopTime.hasTimepoint() && stopTime.timepoint().equals(GtfsStopTimeTimepoint.EXACT);
   }
 
   /**
    * Timepoint without time
    *
-   * <p>Severity: {@code SeverityLevel.WARNING}
+   * <p>Severity: {@code SeverityLevel.WARNING} - to be upgraded as {@code SeverityLevel.ERROR}
```

Comment:
Is there an issue or other link you could reference here for more information about how/why this will be upgraded? Would be nice to have a link to some context in the code itself.